source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -156,6 +156,39 @@ Filters are useful when you need an immediate response that directly modifies th
+
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+|                                                                    | **Open edX Events and Filters**                                                                               | **Forking**                                                                                                                        |
++====================================================================+===============================================================================================================+====================================================================================================================================+
+| **Purpose**                                                        | Allow extending functionality of the Open edX platform without modifying the                                  | Modify the code directly to meet specific needs.                                                                                   |
+|                                                                    | core codebase but by leveraging plugins and configurations.                                                   |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Advantages**                                                     | - More decoupled implementations.                                                                             | - Full control of the platform behavior.                                                                                           |
+|                                                                    | - Decreased maintenance burden of the platform, maintenance efforts are centered on plugins.                  | - No restrictions from what can be changed.                                                                                        |
+|                                                                    | - Enables reusable and configurable behavior across the platform.                                             | - Suitable for highly customized use cases.                                                                                        |
```

Comment:
```suggestion
|                                                                    | - Enables reusable and configurable behavior across the platform.                                             | - Suitable for extremely customized use cases.                                                                                        |
```
I think that hooks framework already allows for high customization. IMHO the modification must be extreme to warrant forking where hooks can do it. The more regular reason is when the platform does not support hooks in the place you want to customize.

Reply:
Another point in this line would be:

> Enables reusable and configurable behavior across the platform

I think is worth splitting in two:

- Enables reusable behavior across the ecosystem and community.
- Enables configurable behavior across the platform

Reply:
> I think that hooks framework already allows for high customization. IMHO the modification must be extreme to warrant forking where hooks can do it. The more regular reason is when the platform does not support hooks in the place you want to customize.

I agree. I added a note suggesting to look into how the framework can help before forking: https://github.com/openedx/docs.openedx.org/pull/617/commits/67bd6f7c4060421d7b2dea42c6affffa8a8d6acd#diff-01d0bcd5d9113aaa8a7f3fde94c7e5ca100265eebebbad867dbf4de21123d451R193

I also split the sentence you mentioned into two for more clarity: https://github.com/openedx/docs.openedx.org/pull/617/commits/67bd6f7c4060421d7b2dea42c6affffa8a8d6acd#diff-01d0bcd5d9113aaa8a7f3fde94c7e5ca100265eebebbad867dbf4de21123d451R172-R173